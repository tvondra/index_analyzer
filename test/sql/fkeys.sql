\set ECHO 0
\i sql/index_analyzer.sql
\set ECHO all

-- can't be executed in a trasaction because we need stats to be updated

-- create table with an indexes, fill it with enough data
CREATE TABLE main_table(col_a INT PRIMARY KEY);
CREATE TABLE child_table(col_a INT, col_b INT REFERENCES main_table(col_a));

-- run basic test on the main table - nothing should be reported
SELECT * FROM analyze_fks('main_table'::regclass);

-- run basic test on the child table - a missing index should be reported
SELECT * FROM analyze_fks('child_table'::regclass);

-- create index on the child table, rerun the check
CREATE INDEX child_idx ON child_table(col_b);

-- run basic test on the child table - nothing should be reported
SELECT * FROM analyze_fks('child_table'::regclass);

-- perform CLEANUP
DROP TABLE child_table;
DROP TABLE main_table;

