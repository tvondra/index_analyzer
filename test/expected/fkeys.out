\set ECHO 0
-- can't be executed in a trasaction because we need stats to be updated
-- create table with an indexes, fill it with enough data
CREATE TABLE main_table(col_a INT PRIMARY KEY);
NOTICE:  CREATE TABLE / PRIMARY KEY will create implicit index "main_table_pkey" for table "main_table"
CREATE TABLE child_table(col_a INT, col_b INT REFERENCES main_table(col_a));
-- run basic test on the main table - nothing should be reported
SELECT * FROM analyze_fks('main_table'::regclass);
 relname | refrelname | fkname | result 
---------+------------+--------+--------
(0 rows)

-- run basic test on the child table - a missing index should be reported
SELECT * FROM analyze_fks('child_table'::regclass);
   relname   | refrelname |         fkname         |           result           
-------------+------------+------------------------+----------------------------
 child_table | main_table | child_table_col_b_fkey | no index on the FK columns
(1 row)

-- create index on the child table, rerun the check
CREATE INDEX child_idx ON child_table(col_b);
-- run basic test on the child table - nothing should be reported
SELECT * FROM analyze_fks('child_table'::regclass);
 relname | refrelname | fkname | result 
---------+------------+--------+--------
(0 rows)

